compiler/rustc_privacy/src/lib.rs

```diff
@@ -1193,12 +1193,14 @@ impl<'tcx> TypePrivacyVisitor<'tcx> {
     fn check_expr_pat_type(&mut self, id: hir::HirId, span: Span) -> bool {
         self.span = span;
         let typeck_results = self.typeck_results();
-        if self.visit(typeck_results.node_type(id)) || self.visit(typeck_results.node_substs(id)) {
+        if self.visit(typeck_results.node_type(id)) == ControlFlow::BREAK
```

Comment:
maybe use a `try` block here instead of having to manually check for `ControlFlow::BREAK` here

Reply:
You mean something like?

```rust
if try {
    self.visit(typeck_results.node_type(id))?;
    self.visit(typeck_results.node_substs(id))
} == ControlFlow::BREAK {
    // ...
}
```

Reply:
yeah, though probably for the whole body of `check_expr_pat_type` as there are quite a lot of these `if`s